scripts/measures/import-qcdr-measures.js

```diff
@@ -120,15 +117,86 @@ const convertCsvToMeasures = function(records, config) {
 
     // If the 'proportion' column (col 17) is Y and the other two columns
     // (continuous and ratio, cols 18 and 19) are N, metricType should be
-    // 'singlePerformanceRate'. Otherwise it should be 'nonProportion'
-    //
-    // Note: if the 'proportion' column is Y *and* there are multiple
-    // strata, then the metricType should be 'multiPerformanceRate'
-    // TODO(kalvin): implement multiPerformanceRate;
-    if (record[17] === 'Y' &&
-        record[18] === 'N' &&
-        record[19] === 'N') {
-      newMeasure['metricType'] = 'singlePerformanceRate';
+    // 'singlePerformanceRate', or 'multiPerformanceRate' if there are multiple
+    // strata/performance rates. Otherwise it should be 'nonProportion'
+    const proportion = _.trim(record[17]);
+    const continuous = _.trim(record[18]);
+    const ratio = _.trim(record[19]);
+    if (proportion === 'Y' && continuous === 'N' && ratio === 'N') {
+      // returns an integer if passed string '3', NaN if passed 'N/A'
+      const numPerformanceRates = _.parseInt(_.trim(record[11]));
+      if (_.isInteger(numPerformanceRates) && numPerformanceRates > 1) {
+        newMeasure['metricType'] = 'multiPerformanceRate';
+
+        const overallPerformanceRate = _.lowerCase(_.trim(record[12]));
+        const nthPerformanceRate = _.parseInt(overallPerformanceRate);
+        if (_.isInteger(nthPerformanceRate)) {
+          newMeasure['overallAlgorithm'] = 'overallStratumOnly';
+        } else if (overallPerformanceRate === 'sum numerators') {
+          newMeasure['overallAlgorithm'] = 'sumNumerators';
+        } else if (overallPerformanceRate === 'weighted average') {
+          newMeasure['overallAlgorithm'] = 'weightedAverage';
+        }
+
+        // Add the names and descriptions of strata
+        let strataName;
+        const measureId = _.trim(record[2]);
+        const measureDescription = _.trim(record[4]);
+
+        // Measure description column contains performance rate description
+        // Split '*summary* Rate 1: text Rate 2: text' into [text, text]
+        const strata = _.split(measureDescription, /\s*[Rr]ate [0-9]+:\s*/);
+        // Drop anything before 'Rate 1' (usually a description of the measure)
+        strata.shift();
+
+        // TODO(kalvin): move strata names to a separate file
+        const STRATA_NAMES = {
```

Comment:
Small batch measure names :)

Reply:
Maybe have a comment indicating that the order of these arrays in important?

Reply:
good idea k